hack/verify-godeps.sh

```diff
@@ -39,9 +39,15 @@ preload-dep() {
 KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 source "${KUBE_ROOT}/hack/lib/init.sh"
 
-branch="${1:-master}"
+readonly branch="origin/${1:-${KUBE_VERIFY_GIT_BRANCH:-master}}"
+echo "Checking for Godeps changes against ${branch}"
+# make sure the branch is valid, otherwise the next check will pass erroneously.
+if ! git describe "${branch}" >/dev/null; then
+  exit 1
+fi
 # notice this uses ... to find the first shared ancestor
-if ! git diff origin/"${branch}"...HEAD | grep 'Godeps/' > /dev/null; then
+if ! git diff --name-only "${branch}...HEAD" | grep 'Godeps/' > /dev/null; then
```

Comment:
also, these checks will never run on post-commit Jenkins, since by definition there's no difference between what's checked out and what's in the branch.

we could maybe add an option to always run these checks on Jenkins. it'd be even better to detect when there have been Godeps changes since the last build, but that's probably a lot harder.